backend/storj/fs.go

```diff
@@ -22,32 +22,96 @@ import (
 	"storj.io/uplink"
 )
 
+var satMap = map[string]string{
+	"us-central-1.tardigrade.io":  "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S@us-central-1.tardigrade.io:7777",
+	"europe-west-1.tardigrade.io": "12L9ZFwhzVpuEKMUNUqkaTLGzwY9G24tbiigLiXpmZWKwmcNDDs@europe-west-1.tardigrade.io:7777",
+	"asia-east-1.tardigrade.io":   "121RTSDpyNZVcEU84Ticf2L1ntiuUimbWgfATz21tuvgk3vzoA6@asia-east-1.tardigrade.io:7777",
+}
+
 // Register with Fs
 func init() {
 	fs.Register(&fs.RegInfo{
 		Name:        "storj",
 		Description: "Storj Decentralized Cloud Storage",
 		NewFs:       NewFs,
+		Config: func(name string, configMapper configmap.Mapper) {
+			satelliteString, _ := configMapper.Get("satellite-address")
+			apiKey, _ := configMapper.Get("api-key")
+			passphrase, _ := configMapper.Get("passphrase")
+			accessString, _ := configMapper.Get("access")
+
+			config.FileDeleteKey(name, "satellite-address")
+			config.FileDeleteKey(name, "api-key")
+			config.FileDeleteKey(name, "passphrase")
+			config.FileDeleteKey(name, fs.ConfigProvider)
+
+			if accessString != "" {
+				return
+			}
+
+			satellite, found := satMap[satelliteString]
+			if !found {
+				satellite = satelliteString
+			}
+
+			access, err := uplink.RequestAccessWithPassphrase(context.TODO(), satellite, apiKey, passphrase)
+			if err != nil {
+				fs.Errorf(nil, "Couldn't create access grant: %v", err)
+				return
+			}
+
+			serialziedAccess, err := access.Serialize()
+			if err != nil {
+				fs.Errorf(nil, "Couldn't serialize access grant: %v", err)
+				return
+			}
+
+			configMapper.Set("access", serialziedAccess)
```

Comment:
Does this persist the change? We were using `config.SetValueAndSave` before to ensure it was written to disk.

Reply:
Maybe the code that calls this ensures it is written?

Reply:
yes, its persistent. At least at the end of the configuration. I never saw usage of `config.SetValueAndSave`.